src/js/shoppingCart.mjs

```diff
@@ -42,4 +50,28 @@ function calculateTotal(cart) {
   });
   document.querySelector(".cart-footer").classList.remove("hide");
   return total;
+}
+
+function addClickEvents(cart) {
+  cart.forEach((cartItem) => {
+    document
+    .getElementById(cartItem.Id)
+    .addEventListener("click", removeItem);
+  });
+}
+
+async function removeItem(e) {
+  let product = await findProductById(e.target.id);
+  let cart = getLocalStorage("so-cart") ?? [];
+  let itemIndex = cart.findIndex((item) => item.Id === product.Id);
+  cart.splice(itemIndex, 1);
+  setLocalStorage("so-cart", cart);
+  //reload page beacuse shopping cart is duplicating items
+  location.reload();
+  //shoppingCart();
```

Comment:
I'm not a fan of reloading the entire page when you remove an item from the cart, but when I was trying to just render the cart items again it was duplicating them and I couldn't figure out why. But reloading the page does work.

Reply:
The cart is being populated using renderListWithTemplate() which adds the cart items as children to the selected element (which starts out empty on a page reload). It's duplicating because it's just inserting another set at the top. If you use the "clear" argument when you call the function, it will empty the element before repopulating it. 

The other option is to use unique identifiers for each cart item and remove them directly from the DOM. This would reduce refreshes, but it would require a lot more logic to select and modify the DOM. Using the cart render function could also help with Mandi's ticket to adjust cart quantities in the cart.

You could either call renderListWithTemplate() manually, or adjust the cart rendering function (something like renderCartContents ?) used on page load to always clear first. We can call it whenever and it should just update the cart based on the local storage.